common/components/org.wso2.carbon.data.agents.log4j/src/main/java/org/wso2/carbon/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -0,0 +1,411 @@
+/*
```

Comment:
Hi Ruwan,

So far we have done the following fixes to do improvement to gain the publisher connectivity with the DAS instance,

> We are not publishing the logs until carbon platform loading the system variable.
> Once the system is ready for log publishing task, appender verify the DAS data publisher is properly initialize, if not it call the publisher init method and  there is no log publishing task are triggered unlit the publisher is available to use.
> Log publisher task use databridge.agent.DataPublisher, since the publisher task is govern by the  DataPublisher API, including the thrift publisher exception. So DAS node shutdown still this publisher didn't get notifications about the log publishing status.

But we are fixed the other blocking issues in following commits:

https://github.com/wso2/shared-analytics/pull/20
https://github.com/wso2/shared-analytics/pull/21
https://github.com/wso2/shared-analytics/pull/99
https://github.com/wso2/shared-analytics/pull/113

Thanks,
SajithD